ext/standard/array.c

```diff
@@ -5902,65 +5902,73 @@ PHP_FUNCTION(array_rand)
 }
 /* }}} */
 
-/* {{{ Returns the sum of the array entries */
-PHP_FUNCTION(array_sum)
+/* Wrapper for array_sum and array_product */
+static void php_array_binop(INTERNAL_FUNCTION_PARAMETERS, const char *op_name, binary_op_type op, zend_long initial)
 {
-	zval *input,
-		 *entry,
-		 entry_n;
+	HashTable *input;
+	zval *entry;
 
 	ZEND_PARSE_PARAMETERS_START(1, 1)
-		Z_PARAM_ARRAY(input)
+		Z_PARAM_ARRAY_HT(input)
 	ZEND_PARSE_PARAMETERS_END();
 
-	ZVAL_LONG(return_value, 0);
+	if (zend_hash_num_elements(input) == 0) {
+		php_error_docref(NULL, E_DEPRECATED, "Passing an empty array is deprecated");
```

Comment:
As a user, I'm a bit concerned about this deprecation.
It seems to me that an empty product or empty sum should be valid, just like it would be the case when using `array_reduce`.

Reply:
ACK will change